### ISSUE TITLE
Storage Access Permission Lost

### ISSUE THREAD

**Issue by dennischen** [colaorange/daily-money-one-doc#31]
On Samsung phones, first obtain Storage permission. After minimizing the program and keeping it in the background for a period of time (1-2 hours), bring it back to the foreground and return to the Data Maintenance Screen. The permission is not available.
```
2024-05-17 19:59:54.854 22591-22591 ViewRootIm...nActivity] com.colaorange.dailymoneyone         I  reportDrawFinished seqId=0 mSyncId=-1 fn=1 mSurfaceChangedTransaction=0x750ca71b00
2024-05-17 19:59:54.859 22591-22591 TaskService             com.colaorange.dailymoneyone         I  Handling intent with task name 'backupDatabaseTask' and app scoping identifier 'com.colaorange.dailymoneyone'.
2024-05-17 19:59:54.920 22591-22591 ViewRootIm...nActivity] com.colaorange.dailymoneyone         I  MSG_WINDOW_FOCUS_CHANGED 1 0
2024-05-17 19:59:54.920 22591-22591 ViewRootIm...nActivity] com.colaorange.dailymoneyone         I  mThreadedRenderer.initializeIfNeeded()#2 mSurface={isValid=true 0x72f89b8000}
2024-05-17 19:59:54.921 22591-22591 InputMethodManager      com.colaorange.dailymoneyone         I  startInputInner - mService.startInputOrWindowGainedFocus
2024-05-17 19:59:54.944 22591-22591 TaskService             com.colaorange.dailymoneyone         I  Handling job with task name 'backupDatabaseTask' for app with scoping identifier 'com.colaorange.dailymoneyone'.
2024-05-17 19:59:55.313 22591-22653 DocumentFile            com.colaorange.dailymoneyone         W  Failed query: java.lang.UnsupportedOperationException: Unsupported Uri content://com.android.externalstorage.documents/tree/primary%3ADMO
2024-05-17 19:59:55.336 22591-22653 DocumentFile            com.colaorange.dailymoneyone         W  Failed query: java.lang.UnsupportedOperationException: Unsupported Uri content://com.android.externalstorage.documents/tree/primary%3ADMO
2024-05-17 19:59:55.386 22591-22659 ReactNativeJS           com.colaorange.dailymoneyone         I  Start to backup database automatically, auto upload config to true
2024-05-17 19:59:55.406 22591-22653 DocumentFile            com.colaorange.dailymoneyone         W  Failed query: java.lang.UnsupportedOperationException: Unsupported Uri content://com.android.externalstorage.documents/tree/primary%3ADMO
2024-05-17 19:59:55.417 22591-22653 DocumentFile            com.colaorange.dailymoneyone         W  Failed query: java.lang.UnsupportedOperationException: Unsupported Uri content://com.android.externalstorage.documents/tree/primary%3ADMO
2024-05-17 19:59:55.427 22591-22653 DocumentFile            com.colaorange.dailymoneyone         W  Failed query: java.lang.UnsupportedOperationException: Unsupported Uri content://com.android.externalstorage.documents/tree/primary%3ADMO
2024-05-17 19:59:55.657 22591-22659 ReactNativeJS           com.colaorange.dailymoneyone         I  Backed up to content://com.android.externalstorage.documents/tree/primary%3ADMO/document/primary%3ADMO%2Fauto-dmo-20240517-1959.db
2024-05-17 19:59:58.491 22591-22599 e.dailymoneyone         com.colaorange.dailymoneyone         I  Background concurrent copying GC freed 90446(4055KB) AllocSpace objects, 36(9896KB) LOS objects, 49% free, 14MB/29MB, paused 456us,477us total 160.892ms
2024-05-17 19:59:58.491 22591-22599 e.dailymoneyone         com.colaorange.dailymoneyone         W  Reducing the number of considered missed Gc histogram windows from 683 to 100
2024-05-17 19:59:58.524 22591-22601 System                  com.colaorange.dailymoneyone         W  A resource failed to call close. 
2024-05-17 19:59:58.587 22591-22591 ViewRootIm...nActivity] com.colaorange.dailymoneyone         I  ViewPostIme pointer 0
2024-05-17 19:59:58.620 22591-22591 ViewRootIm...nActivity] com.colaorange.dailymoneyone         I  ViewPostIme pointer 1
```

**Comment by dennischen** [comment]
more logcat
'''
2024-05-18 21:54:37.840 31455-31534 ReactNativeJS           com.colaorange.dailymoneyone         E  'runAsync', [Error: Cannot access realm that has been closed.]
2024-05-18 21:54:37.843 31455-31535 unknown:ReactNative     com.colaorange.dailymoneyone         E  console.error: runAsync Error: Cannot access realm that has been closed., js engine: hermes, stack:
                                                                                                    _construct@1:104413
                                                                                                    Wrapper@1:104067
                                                                                                    _callSuper@1:102052
                                                                                                    SyntheticError@1:103544
                                                                                                    reactConsoleErrorHandler@1:103205
                                                                                                    logError@1:1200051
                                                                                                    ?anon_0_@1:1117113
                                                                                                    asyncGeneratorStep@1:541174
                                                                                                    _throw@1:541476
                                                                                                    tryCallOne@53:15
                                                                                                    anonymous@139:26
                                                                                                    anonymous@1:121459
                                                                                                    _callTimer@1:120455
                                                                                                    _callReactNativeMicrotasksPass@1:120600
                                                                                                    callReactNativeMicrotasks@1:122536
                                                                                                    __callReactNativeMicrotasks@1:87997
                                                                                                    anonymous@1:87130
                                                                                                    __guard@1:87871
                                                                                                    flushedQueue@1:87041
'''

**Comment by dennischen** [comment]
It is related to issue #30.

**Comment by dennischen** [comment]
Fixed with #30
